### PR TITLE
Fix / TAO-6994 Cover cases like .5 float input; 2 tests for that

### PR DESCRIPTION
**Pull request by oatymart**
The bug arose because we are using our own locale-specific `parseFloat` function in `util/locale.js`. 

There was an edge case which was missed: splitting a string which begins with the decimal separator -> the part before the separator will end up empty. The first part will now become 0 in this case.

### CODE REVIEW

**Comment by oatymart** [comment]
This rewrite of `locale.parseFloat` handles negative float inputs as well as omission of the leading zero.

Please note that the default locale for TAO is set in `config/tao/client_lib_config_registry.conf.php` as
```
    'config' => array(
        'util/locale' => array(
            'decimalSeparator' => '.',
            'thousandsSeparator' => '',
            'dateTimeFormat' => 'DD/MM/YYYY HH:mm:ss'
        ),
```
In other words we are currently set up to use a decimal point and no thousands separator.

If a user or client chooses to modify this configuration, the new code will handle whichever separators they specify here.

File: views/js/util/locale.js [hunk 1]
```diff
@@ -73,22 +73,23 @@ define(['module', 'moment'], function (module, moment) {
...
             if (this.getThousandsSeparator().length) {
-                ones = parts[0].replace(new RegExp('\\' + this.getThousandsSeparator(), 'g'), '');
+                numStr = numStr.replace(new RegExp('\\' + this.getThousandsSeparator(), 'g'), '');
             }
-
-            return parseFloat(ones) + parseFloat('0.' + parts[1]);
+        
+            // standardise the decimal separator as '.':
+            if (this.getDecimalSeparator() != '.') {
+                numStr = numStr.replace(new RegExp('\\' + '.', 'g'), '_')
```
Comment (dietertaotesting) [inline views/js/util/locale.js]: type comparison required `this.getDecimalSeparator() !== '.'`

Comment (oatymart) [reply views/js/util/locale.js]: Changed.
